cmd/reference-addon-manager/main.go

```diff
@@ -92,11 +93,40 @@ func main() {
 		}
 	}
 
+	// fetching the following from the Downwards Api in case of reference-addon.
+	// Can be hardcoded or supplied in any other way too depending on the addon developers.
+	addonName := os.Getenv("ADDON_NAME")
+	addonNamespace := os.Getenv("ADDON_NAMESPACE")
+
+	// Setup the StatusReporter
+	addonSdkClient := NewAddonSDKClient(mgr.GetClient())
+	statusReporter := addonsdk.SetupStatusReporter(addonSdkClient, addonName, addonNamespace, ctrl.Log.WithName("StatusReporter"))
+	if err != nil {
+		setupLog.Error(err, "unable to setup status-reporter")
+		os.Exit(1)
+	}
+
+	if err := mgr.Add(statusReporter); err != nil {
+		setupLog.Error(err, "unable to start status-reporter")
```

Comment:
`mgr.Add()` is not starting the statusReporter, it's just registering it into the manager to start later when `mgr.Start()` is called.
```suggestion
		setupLog.Error(err, "unable to add status-reporter to manager")
```

Reply:
Whoops!